File: csharp/DVSClient/DVSClient/server/address/search/RestApiSearchResultSuggestion.cs
using Newtonsoft.Json;

namespace DVSClient.Server.Address.Search
{
    public class RestApiSearchResultSuggestion
    {
        [JsonProperty("global_address_key")]
        public string? GlobalAddressKey { get; set; }

        [JsonProperty("text")]
        public string? Text { get; set; }

        [JsonProperty("matched")]
        public IEnumerable<IEnumerable<int>>? Matched { get; set; }

        [JsonProperty("format")]
        public string? FormatUrl { get; set; }

        [JsonProperty("dataset")]
        public string? Dataset { get; set; }

        [JsonProperty("additional_attributes")]
        public IEnumerable<RestApiSearchResultAdditionalAttribute>? AdditionalAttributes { get; set; }
    }
}

File: typescript/src/server/address/search/restApiSearchResultSuggestion.ts
import { RestApiSearchResultAdditionalAttribute } from "./restApiSearchResultAdditionalAttribute";

export type RestApiSearchResultSuggestion = {
    global_address_key?: string;
    text?: string;
    matched?: number[][],
    format?: string;
    dataset?: string;
    additional_attributes?: RestApiSearchResultAdditionalAttribute[]
}
